static/js/composebox_typeahead.js

```diff
@@ -775,108 +806,140 @@ exports.content_typeahead_selected = function (item, event) {
-        } else {
-            // If more text after the input, then complete the token, but don't touch
-            // "rest" (i.e. do not add a closing fence)
-            beginning = beginning.slice(0, backticks) + item;
-        }
-    } else if (this.completing === "topic_jump") {
-        // Put the cursor at the end of immediately preceding stream mention syntax,
-        // just before where the `**` at the end of the syntax.  This will delete that
-        // final ** and set things up for the topic_list typeahead.
-        const index = beginning.lastIndexOf("**");
-        if (index !== -1) {
-            beginning = beginning.slice(0, index) + ">";
+        case "topic_jump": {
+            // Put the cursor at the end of immediately preceding stream mention syntax,
+            // just before where the `**` at the end of the syntax.  This will delete that
+            // final ** and set things up for the topic_list typeahead.
+            const index = beginning.lastIndexOf("**");
+            if (index !== -1) {
+                beginning = beginning.slice(0, index) + ">";
+            }
+            break;
         }
-    } else if (this.completing === "topic_list") {
-        // Stream + topic mention typeahead; close the stream+topic mention syntax
-        // with the topic and the final **.
-        const start = -this.token.length;
-        beginning = beginning.slice(0, start) + item + "** ";
-    } else if (this.completing === "time_jump") {
-        let timestring = beginning.slice(Math.max(0, beginning.lastIndexOf("<time:")));
-        if (timestring.startsWith("<time:") && timestring.endsWith(">")) {
-            timestring = timestring.slice(6, -1);
+        case "topic_list": {
+            // Stream + topic mention typeahead; close the stream+topic mention syntax
+            // with the topic and the final **.
+            const start = -this.token.length;
+            beginning = beginning.slice(0, start) + item + "** ";
+            break;
         }
-        const timestamp = timerender.get_timestamp_for_flatpickr(timestring);
-
-        const on_timestamp_selection = (val) => {
-            const datestr = val;
-            beginning =
-                beginning.slice(0, Math.max(0, beginning.lastIndexOf("<time"))) +
-                `<time:${datestr}> `;
-            if (rest.startsWith(">")) {
-                rest = rest.slice(1);
+        case "time_jump": {
+            let timestring = beginning.slice(Math.max(0, beginning.lastIndexOf("<time:")));
+            if (timestring.startsWith("<time:") && timestring.endsWith(">")) {
+                timestring = timestring.slice(6, -1);
             }
-            textbox.val(beginning + rest);
-            textbox.caret(beginning.length, beginning.length);
-            compose_ui.autosize_textarea(textbox);
-        };
-        show_flatpickr(this.$element[0], on_timestamp_selection, timestamp);
-        return beginning + rest;
```

Comment:
@aero31aero I assume it's intentional that this case doesn't fall through -- should we have a comment on that?  It's unique for this case to not fall through to the common code with the `setTimeout` below.

Reply:
@aero31aero ping on this code history question.